inst/htmlwidgets/leaflet.js

```diff
@@ -856,6 +858,36 @@ var dataframe = (function() {
     });
   };
 
+		// TO DO: right now using .id but may let users choose, let users choose "<" or ">" or "=" (default)
+		// consider letting users specify listed style arguments ala addGeoJSON
+		// and create methods for many ids to style or remove with single loop, and many ids to many styles
```

Comment:
Yes it's slow. We really need a way to send over setStyle or resetStyle functions to the client, although I'm not sure yet how to do that and allow customization. Note: I don't believe resetStyle will work for when you effectively want to change back to the prior style instead of the originally loaded style (when you have dynamic data). Internally, on `leaflet.js` we have hard-coded:

```javascript
    function highlightFeature(e) {
    	var layer = e.target;
			layer.setStyle({weight: 3,color: "black"});
	if (!L.Browser.ie && !L.Browser.opera) {
		layer.bringToFront();};
    };
    function resetHighlight(e) {
      var layer = e.target;
      var originalColor = e.target.feature.properties.style.color;
      var originalWeight = e.target.feature.properties.style.weight;
      layer.setStyle({weight: originalWeight,color: originalColor});
    };
```

and inside the `onEachFeature` of `var gjlayer` of `methods.addGeoJSON` we include:

```javascript
    layer.on("mouseover", highlightFeature , this);
    layer.on("mouseout", resetHighlight, this);
```

The capability of assigning styles to events could probably be added to `addGeoJSON`. Then, we probably need some sort of function `styleGeoJSON` that lets you restyle an entire layer (many users won't actually have dynamic data), and a function `styleEventGeoJSON` that lets you restyle the event styles .